### PR TITLE
[SD-4312] Preserve desk and stage in routing

### PR DESCRIPTION
**Pull request by marwoodandrew**

### CODE REVIEW

File: superdesk/io/feed_parsers/nitf.py [hunk 1]
```diff
@@ -146,6 +142,18 @@ def parse_meta(tree, item):
...
+    desk_name = tree.find('head/meta[@name="aap-desk"]')
+    if desk_name is not None:
+        desk = superdesk.get_resource_service('desks').find_one(req=None, name=desk_name.get('content'))
+        if desk:
+            item['task'] = {'desk': desk.get('_id')}
+            stage_name = tree.find('head/meta[@name="aap-stage"]')
+            if stage_name is not None:
+                lookup = {'$and': [{'name': stage_name.get('content')}, {'desk': str(desk.get('_id'))}]}
+                stages = superdesk.get_resource_service('stages').get(req=None, lookup=lookup)
+                for stage in stages:
```
Comment (akintolga) [inline superdesk/io/feed_parsers/nitf.py]: Here we expect to find 0 or 1 item, maybe for loop is not needed...

File: apps/rules/routing_rules.py [hunk 2]
```diff
@@ -210,7 +210,12 @@ def apply_routing_scheme(self, ingest_item, provider, routing_scheme):
             if filters_service.does_match(content_filter, ingest_item):
                 if rule.get('actions', {}).get('preserve_desk', False) and ingest_item.get('task', {}).get('desk'):
                     desk = get_resource_service('desks').find_one(req=None, _id=ingest_item['task']['desk'])
-                    self.__fetch(ingest_item, [{'desk': desk[config.ID_FIELD], 'stage': desk['incoming_stage']}])
+                    if ingest_item.get('task', {}).get('stage'):
+                        stage = get_resource_service('stages').find_one(req=None, _id=ingest_item['task']['stage'])
```
Comment (akintolga) [inline apps/rules/routing_rules.py]: Is this line needed? We already know the stage id from ingest_item['task']['stage']...

Comment (marwoodandrew) [inline apps/rules/routing_rules.py]: True

**Comment by akintolga** [comment]
:+1:

**Comment by mdhaman** [comment]
:+1:

**Comment by petrjasek** [comment]
:+1:

**Pull request merged**